### PR TITLE
Change to scope file tree to the known path if root folders dont return correctly

### PR DESCRIPTION
**Pull request by udeeshagautam**
Aims to fix https://github.com/microsoft/azuredatastudio/issues/4767 : 
Issue : File browser for back up result does not load when the root folders return are not valid and hence the expand path i.e. the default backup path can not be found in those. 
Fix: workaround to scope the file browser tree to the default backup path

Following is how it looks when root not found. (Please note that if there is nothing present in the default folder it will show as empty)

backup folder

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs [hunk 1]
```diff
@@ -248,6 +260,26 @@ internal bool FilterFile(string fileName, string[] masks)
...
+            this.fileTree = new FileTree();
+            FileTreeNode node = new FileTreeNode()
+            {
+                Name = expandPath,
+                FullPath = expandPath,
+                IsExpanded = true
+            };
+
+            this.fileTree.RootNode.AddChildNode(node);
+            node.Children = node.Children = this.GetChildren(node.FullPath);
```
Comment (shueybubbles) [inline src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs]: duplicate assignment of node.Children

File: src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs [hunk 2]
```diff
@@ -248,6 +260,26 @@ internal bool FilterFile(string fileName, string[] masks)
...
+        {
+            this.fileTree = new FileTree();
+            FileTreeNode node = new FileTreeNode()
+            {
+                Name = expandPath,
+                FullPath = expandPath,
+                IsExpanded = true
+            };
+
+            this.fileTree.RootNode.AddChildNode(node);
```
Comment (shueybubbles) [inline src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs]: is this adding '/var/opt/mssql/data' as a child node of the root in addition to any other nodes that came back from enumerating the root?

**Comment by shueybubbles** [comment]
Can you add a screenshot?

File: src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs (hunk 2 above)
Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs]: /var/opt/mssql/data is the this.fileTree.RootNode. This is adding the bak files and any child folders to it.

**Comment by kburtram** [comment]
@udeeshagautam do you know what happened with the code coverage?  perhaps some tests are failing?

**Comment by udeeshagautam** [comment]
@kburtram yes, the last run had -1% decline.. so I re-ran and it went -4.5%.
I see quite a few "Login failed for user 'sa'." - not related to this area as such but will take a look before merging...

**Comment by udeeshagautam** [comment]
@shueybubbles @chlafreniere would you like to take a look or I should close?

File: src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs [hunk 3]
```diff
@@ -160,7 +160,19 @@ public void ExpandSelectedNode(string expandPath)
...
+
+                        // If root folders returned from sys.dm_os_enumerate_filesystem don't match the folders present, expandPath might not be found
+                        // Bug: https://github.com/microsoft/azuredatastudio/issues/4767
+                        // Workaround : scope file browser tree to expandPath explicitly 
+                        try
+                        {
+                            this.ScopeFileTreeToPath(expandPath);
+                            return;
+                        }
+                        catch
```
Comment (ranasaria) [inline src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs]: What is the type of Exception that we are replacing with the FileBrowserException? Should we wrap the exception caught as an inner exception of log the details of the exception that we are swallowing?

Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs]: Yes that would be good idea, however in file browser service the result only seems to pick the message from exception as result.Message = ex.Message; so it wouldn't matter. I can take the change of enabling passing on the whole exception as a separate change overall.

**Pull request merged**